### PR TITLE
Remove weak/strong self dances

### PR DESCRIPTION
**Pull request by 0xced**
The weak/strong self dances are unnecessary here, see [this Stack Overflow answer][1] to understand why.

[1]: http://stackoverflow.com/questions/19018456/ios-blocks-and-strong-weak-references-to-self/19018633#19018633

### CODE REVIEW

**Comment by smileyborg** [comment]
Yeah, Cédric is correct, since the block isn't going to be retained by `self` in any case here, it's unnecessary (although not harmful).

**Comment by radazzouz** [comment]
I agreed with @smileyborg that the `weak/strong` self dance is unnecessary and not harmful. The dance was added in the early days of iOS extensions when we weren't too sure about the consequences of the peculiar life cycle of iOS extensions.

I do not think that we can merge the PR though. I think that the documentation should reflect whatever's in the code. If we wish to remove `weak/strong` self dance from `README` we also need to remove them from both **ACME** and **ACME Browser** and test it thoroughly before merging to master.

I will close this PR for now. @0xced, can you please open a new PR with the all the necessary?

Cheers!

**Comment by 0xced** [comment]
I updated the pull request and removed the weak/strong self dances everywhere.

File: README.md [hunk 1]
```diff
@@ -212,10 +204,9 @@ Adding 1Password to your change password screen is very similar to adding 1Passw
 			return;
 		}
 
-		__strong typeof(self) strongMe = miniMe;
-		strongMe.oldPasswordTextField.text = loginDict[AppExtensionOldPasswordKey];
-		strongMe.freshPasswordTextField.text = loginDict[AppExtensionPasswordKey];
-		strongMe.confirmPasswordTextField.text = loginDict[AppExtensionPasswordKey];
+		self.oldPasswordTextField.text = loginDict[AppExtensionOldPasswordKey];
+		self.freshPasswordTextField.text = loginDict[AppExtensionPasswordKey];
+		self.confirmPasswordTextField.text = loginDict[AppExtensionPasswordKey];
```
Comment (radazzouz) [inline README.md]: Could you please remove the extra new line here?

File: Demos/App Demo for iOS/App Demo for iOS/ChangePasswordViewController.m [hunk 2]
```diff
@@ -75,10 +73,9 @@ - (IBAction)changePasswordIn1Password:(id)sender {
 			return;
 		}
 
-		__strong typeof(self) strongMe = miniMe;
-		strongMe.oldPasswordTextField.text = loginDict[AppExtensionOldPasswordKey];
-		strongMe.freshPasswordTextField.text = loginDict[AppExtensionPasswordKey];
-		strongMe.confirmPasswordTextField.text = loginDict[AppExtensionPasswordKey];
+		self.oldPasswordTextField.text = loginDict[AppExtensionOldPasswordKey];
+		self.freshPasswordTextField.text = loginDict[AppExtensionPasswordKey];
+		self.confirmPasswordTextField.text = loginDict[AppExtensionPasswordKey];
```
Comment (radazzouz) [inline Demos/App Demo for iOS/App Demo for iOS/ChangePasswordViewController.m]: Could you please remove the extra new line here?

**Comment by radazzouz** [comment]
Thanks @0xced! :+1:

File: README.md [hunk 3]
```diff
@@ -212,10 +204,9 @@ Adding 1Password to your change password screen is very similar to adding 1Passw
 			return;
 		}
 
-		__strong typeof(self) strongMe = miniMe;
-		strongMe.oldPasswordTextField.text = loginDict[AppExtensionOldPasswordKey];
-		strongMe.freshPasswordTextField.text = loginDict[AppExtensionPasswordKey];
-		strongMe.confirmPasswordTextField.text = loginDict[AppExtensionPasswordKey];
+		self.oldPasswordTextField.text = loginDict[AppExtensionOldPasswordKey];
+		self.freshPasswordTextField.text = loginDict[AppExtensionPasswordKey];
```
Comment (radazzouz) [inline README.md]: Could you please remove the extra new line here?

Comment (0xced) [inline README.md]: I'm not sure how you are looking at the diff, but there's no extra new line anywhere.

Comment (radazzouz) [inline README.md]: You are right @0xced! The newlines only show up in Github. Everything looks good in Xcode.

**Comment by radazzouz** [comment]
After testing the changes in both ACME and ACME Browser, all seems to work fine.

:+1: @0xced. I am merging this PR.

**Pull request merged**